service/qos/raft_service_level_distributed_data_accessor.cc

```diff
@@ -0,0 +1,101 @@
+    co_return co_await qp.get_mutations_internal(delete_query, qos_query_state(), timestamp, {service_level_name});
+}
+
+raft_service_level_distributed_data_accessor::raft_service_level_distributed_data_accessor(cql3::query_processor& qp, service::raft_group0_client& group0_client)
+    : _qp(qp)
+    , _group0_client(group0_client) {}
+
+future<qos::service_levels_info> raft_service_level_distributed_data_accessor::get_service_levels() const {
+    return qos::get_service_levels(_qp, db::system_keyspace::NAME, db::system_keyspace::SERVICE_LEVELS_V2, db::consistency_level::LOCAL_ONE);
+}
+
+future<qos::service_levels_info> raft_service_level_distributed_data_accessor::get_service_level(sstring service_level_name) const {
+    return qos::get_service_level(_qp, db::system_keyspace::NAME, db::system_keyspace::SERVICE_LEVELS_V2, std::move(service_level_name), db::consistency_level::LOCAL_ONE);
+}
+
+future<> raft_service_level_distributed_data_accessor::do_raft_command(std::optional<service::group0_guard> guard, std::function<future<std::vector<mutation>>(api::timestamp_type)> mutation_function, std::string_view description, std::vector<mutation> additional_mutations) const {
+    if (this_shard_id() != 0) {
+        on_internal_error(logger, "raft_service_level_distributed_data_accessor: must be executed on shard 0");
+    }
+
+    if (!guard) {
+        on_internal_error(logger, "raft_service_level_distributed_data_accessor: guard must be present");
+    }
+    
+    //NOTE: abort_source here is temporary. It's fixed and passed as an argument in a later commit
+    abort_source as;
+    auto timestamp = guard->write_timestamp();
+
+    auto muts = co_await mutation_function(timestamp);
+    if (additional_mutations.size() > 0) {
+        muts.insert(muts.end(), additional_mutations.begin(), additional_mutations.end());
+    }
+    service::write_mutations change {
+        .mutations{muts.begin(), muts.end()},
+    };
+
+    auto group0_cmd = _group0_client.prepare_command(change, *guard, description);
+    co_await _group0_client.add_entry(std::move(group0_cmd), std::move(*guard), &as);
+}
+
+future<> raft_service_level_distributed_data_accessor::set_service_level(sstring service_level_name, qos::service_level_options slo, std::optional<service::group0_guard> guard) const {
+    co_await do_raft_command(std::move(guard), [this, service_level_name, slo] (api::timestamp_type timestamp) {
+        return get_set_mutations(_qp, timestamp, service_level_name, slo);
+    }, "set service level");
+}
+
+future<> raft_service_level_distributed_data_accessor::drop_service_level(sstring service_level_name, std::vector<mutation> remove_attribute_mutation, std::optional<service::group0_guard> guard) const {
+    co_await do_raft_command(std::move(guard), [this, service_level_name] (api::timestamp_type timestamp) {
+        return get_delete_mutations(_qp, timestamp, service_level_name);
```

Comment:
I am not sure if those additional abstractions are needed. I.e. you have get_delete_mutations which implementation is 2 lines and its used only once, here. Why not place them directly here (see https://github.com/scylladb/scylladb/pull/16585/commits/766bfcd03ae0bc8288e80b9b6559be50b8e49922#r1516044594 also)

It can be also a bit confusing because you need an extra hop to understand that delete is doing drop. Or figure out what get_set_mutations is doing (unfortunate name).